### PR TITLE
ntp/config: add generator to filter out the duplicated ntp server

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
related to https://github.com/harvester/harvester/issues/3133#issuecomment-1595486084

### CODE REVIEW

**Review by bk201 — APPROVED** [review]
LGTM

File: pkg/controller/nodeconfig/config/ntp.go [hunk 1]
```diff
@@ -439,3 +442,22 @@ func updateCondition(conditions []nodeconfigv1.ConfigStatus, c nodeconfigv1.Conf
...
 	return conditions
 }
+
+func reGenerateNTPConfig(ntpconfigs *nodeconfigv1.NTPConfig) *nodeconfigv1.NTPConfig {
+	if ntpconfigs.NTPServers == "" {
+		return ntpconfigs
+	}
+
+	// fileter the duplicated NTP servers
+	currentNTPServers := strings.Split(ntpconfigs.NTPServers, " ")
```
Comment (guangbochen) [inline pkg/controller/nodeconfig/config/ntp.go:452]: just wondering why do we prefer " " over the comma as a splitter, is this a UI choice?

Comment (Vicente-Cheng) [reply pkg/controller/nodeconfig/config/ntp.go:452]: Because the NTP servers on the config are split by a ` `, we need to convert it back to list.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**